subsys/net/lib/lwm2m/lwm2m_object.h

```diff
@@ -278,6 +278,7 @@ struct lwm2m_engine_context {
 	struct lwm2m_input_context *in;
 	struct lwm2m_output_context *out;
 	struct lwm2m_obj_path *path;
+	size_t block_offset;
```

Comment:
I saved the block_offset at the lwm2m_engine_context because it will be passed to the "post_write_cb" at lwm2m_engine_write() function.
I could use get_block_ctx() each time when lwm2m_engine_write() is called but I chose to store the block_offset at the context to save the query time.